### PR TITLE
S7-ChannelList

### PR DESCRIPTION
**Pull request by chorfiyoussef**
This pull request (PR) on the "SwEntApp" project introduces significant enhancements to the user messaging and channel navigation features within the application. It focuses on improving user interaction and message management through the following updates:

**Channel List Navigation**: A new feature is added that allows users to see a list of message channels. This is accessible via the icon at the left of the GoMeet search bar. This functionality enables users to easily navigate and switch between different discussion channels, enhancing user engagement and communication efficiency.

**Integration with `io.getstream.chat` Libraries**: The PR integrates more deeply with the Stream Chat libraries, specifically utilizing components like `ChannelsScreen` and `MessagesScreen`. These components are used to build out the UI for displaying channels and messages.

Overall, this PR significantly enhances the messaging functionality of the SwEntApp, making it more interactive and user-friendly by providing users with a clear view of their message channels and easier navigation options within the app.

### CODE REVIEW

**Review by Antoine444 — APPROVED** [review]
This additional feature of the screen with all the discussions of the users with the different users he interacted with makes the app a lot more userfriendly and your implementation is very clean. Would it be possible with the library `io.getstream.chat` to also use the buttons at the top left and top right of the MessagesScreen ? The button on the top left would allow the user to navigate to his profile and the one on the top right would allow the user to select an user in his friends list to chat with.

Screenshot 2024-05-08 at 10 03 03 PM

File: app/src/main/java/com/github/se/gomeet/MainActivity.kt
[the remark this answers is not in the document]
Comment (chorfiyoussef) [reply app/src/main/java/com/github/se/gomeet/MainActivity.kt:1]: Thank you ! `messageLimit = 30` corresponds to the number of messages to load in a single page. This is the value used in the getstream.io tutorial.

**Comment by chorfiyoussef** [comment]
> This additional feature of the screen with all the discussions of the users with the different users he interacted with makes the app a lot more userfriendly and your implementation is very clean. Would it be possible with the library `io.getstream.chat` to also use the buttons at the top left and top right of the MessagesScreen ? The button on the top left would allow the user to navigate to his profile and the one on the top right would allow the user to select an user in his friends list to chat with.
> Screenshot 2024-05-08 at 10 03 03 PM

This suggestion really makes sense and I'll try and see what I can do before merging

File: app/src/main/java/com/github/se/gomeet/MainActivity.kt
[the remark this answers is not in the document]
Comment (Antoine444) [reply app/src/main/java/com/github/se/gomeet/MainActivity.kt:1]: Ok perfect ! And thank you for adding the suggestion I offered, it's great !

**Review by Antoine444 — APPROVED** [review]

**Pull request merged**